app/courses/page.tsx

```diff
@@ -13,17 +14,29 @@ import { Path as PathProps } from "@/api/types";
 import Image from "next/image";
 import Link from "next/link";
 import Card from "@/components/ui/card";
-
 import platypusLogo from "../../public/platypus.svg";
-import { FaLock } from "react-icons/fa";
+import { FaStar } from "react-icons/fa";
 
 // TODO: update "article" to "learning path"
 
 const Paths = async () => {
   const paths = await readPaths();
   // return <pre>{JSON.stringify(paths, null, 2)}</pre>;
+
+  // Sort paths by isFree first and then by level
+  paths.data.sort((a, b) => {
```

Comment:
Few things I would say here, 
1. try not to mutate data like this, create a new const with the new result.
2. the order of this data should be specified over the request i.e. inside `readPaths` in the parameters
3. defining what it means to sort an array is cool, but almost always overkill unless you're in OOP land

That being said, this works (I assume) and it's compute that happens on the server and promptly cached so is not a big deal. If you're happy I'm happy.

Reply:
Here's why mutable code is a bit of a coding smell: https://dev.to/dglsparsons/how-to-write-immutable-code-and-never-get-stuck-debugging-again-4p1